### PR TITLE
Fix FieldDefinition round-trip test for Double

### PR DESCRIPTION
**Pull request by mitchellvitez**
Only generate doubles under 15 digits long, necessary because of a
PostgreSQL limitation

### CODE REVIEW

File: orville-postgresql-libpq/test/Test/FieldDefinition.hs [hunk 1]
```diff
@@ -98,17 +94,15 @@ fieldDefinitionTree pool =
...
 
--- I tried generating doubles limited to 15 digits, but hedgehog simply gave
--- up because this generator never produced anything. I added a story to track
--- this thread and opted to move forward with getting FieldDefinition in.
---
---storableDoubleGen :: HH.Gen Double
---storableDoubleGen =
---  let
---    isStorable :: Double -> Bool
---    isStorable d = floatDigits d <= 15
```
Comment (mitchellvitez) [inline orville-postgresql-libpq/test/Test/FieldDefinition.hs:109]: We could have used a filter like
```
isStorable = (<= 15) . length . filter (/= '.') . takeWhile (/= 'e') . show
```
but I thought it was kind of iffy to rely on `show` for `Double` in that way

Comment (telser) [reply orville-postgresql-libpq/test/Test/FieldDefinition.hs:109]: +1, definitely would prefer to not use `show` like that.

File: orville-postgresql-libpq/test/Test/FieldDefinition.hs [hunk 2]
```diff
@@ -98,17 +94,15 @@ fieldDefinitionTree pool =
...
---    Gen.filter isStorable $ Gen.double (Range.linearFracFrom 0 (-1000) 1000)
+storableDoubleGen :: HH.Gen Double
+storableDoubleGen =
+  let
+    -- Necessary because PostgreSQL only stores up to 15 digits of precision
+    -- With a 3-digit range, this gives us 12 places after the decimal
+    truncateLongDouble :: Double -> Double
+    truncateLongDouble = (/1e12) . (fromIntegral :: Int -> Double) . round . (*1e12)
+  in
+    flip Gen.subterm truncateLongDouble $ Gen.double $ Range.linearFracFrom 0 (-1000) 1000
```
Comment (telser) [inline orville-postgresql-libpq/test/Test/FieldDefinition.hs:105]: That first `$` _could_ be a `.`

**Review by telser — APPROVED** [review]
This is great.

File: orville-postgresql-libpq/test/Test/FieldDefinition.hs (hunk 2 above)
Comment (mitchellvitez) [reply orville-postgresql-libpq/test/Test/FieldDefinition.hs:105]:
```suggestion
    flip Gen.subterm truncateLongDouble . Gen.double $ Range.linearFracFrom 0 (-1000) 1000
```

**Pull request merged**